### PR TITLE
(test):Adding some test coverage for form-helper functions

### PR DESCRIPTION
**Pull request by gitcliff**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
PR is adding test coverage for some form helper functions

### CODE REVIEW

File: src/utils/form-helper.test.ts [hunk 1]
```diff
@@ -80,4 +95,404 @@ describe('Form Engine Helper', () => {
...
+
+      expect(field.readonly).toBe(true);
+    });
+  });
+
+  describe('parseToLocalDateTime', () => {
+    it('should parse valid date string with time correctly', () => {
+      const dateString = '2023-06-27T14:30:00';
+      const expectedDate = new Date(2023, 5, 27, 14, 30, 0);
+      const parsedDate = parseToLocalDateTime(dateString);
```
Comment (brandones) [inline src/utils/form-helper.test.ts:393]: @samuelmale why aren't we using [parseDate](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/docs/API.md#parsedate)?

File: src/utils/form-helper.test.ts [hunk 2]
```diff
@@ -80,4 +95,404 @@ describe('Form Engine Helper', () => {
...
+      const result = evaluateDisabled(node, allFields, allValues, sessionMode, patient, mockExpressionRunnerFn);
+      expect(result).toBe(false);
+      expect(mockExpressionRunnerFn).toHaveBeenCalledWith(
+        node.value.disabled.disableWhenExpression,
+        node,
+        allFields,
+        allValues,
+        { mode: sessionMode, patient },
+      );
+    });
```
Comment (brandones) [inline src/utils/form-helper.test.ts:487]: I cannot imagine what could cause this test to fail

```suggestion
```

**Review by brandones — APPROVED** [review]
Just the one question @gitcliff , otherwise LGTM. Please answer the question (or delete the test) and then we'll merge this in.

I'm not well-positioned to evaluate whether these are actually useful tests, or whether it is too much "testing implementation details." I will leave that evaluation to @samuelmale and @pirupius .

File: src/utils/form-helper.test.ts (hunk 1 above)
Comment (samuelmale) [reply src/utils/form-helper.test.ts:393]: No good reason! I agree we should rely more on the utils exposed by the framework.

File: src/utils/form-helper.test.ts (hunk 2 above)
Comment (pirupius) [reply src/utils/form-helper.test.ts]: Deleted

**Review by pirupius — APPROVED** [review]

**Pull request merged**